sdk/ai/Azure.AI.Inference/tests/Samples/Sample13_ChatCompletionsWithStructuredOutput.cs

```diff
@@ -0,0 +1,154 @@
+                        "prep_time": {
+                            "type": "string"
+                        },
+                        "bake_time": {
+                            "type": "string"
+                        }
+                    }
+                    """) },
+                { "required", BinaryData.FromString("[\"ingredients\", \"steps\", \"bake_time\"]") },
+                { "additionalProperties", BinaryData.FromString("false") }
+            };
+
+            requestOptions.ResponseFormat = ChatCompletionsResponseFormat.CreateJsonFormat("cakeBakingDirections", jsonSchema);
+
+            Response<ChatCompletions> response = client.Complete(requestOptions);
+            #endregion
+
+            Assert.That(response, Is.Not.Null);
+            Assert.That(response.Value, Is.InstanceOf<ChatCompletions>());
+            ChatCompletions result = response.Value;
+            Assert.That(result.Id, Is.Not.Null.Or.Empty);
+            Assert.That(result.Created, Is.Not.Null.Or.Empty);
+            Assert.That(result.FinishReason, Is.EqualTo(CompletionsFinishReason.Stopped));
+            Assert.That(result.Role, Is.EqualTo(ChatRole.Assistant));
+            Assert.That(result.Content, Is.Not.Null.Or.Empty);
+
+            #region Snippet:Azure_AI_Inference_SampleStructuredOutputParseJson
+            using JsonDocument structuredJson = JsonDocument.Parse(result.Content);
+            structuredJson.RootElement.TryGetProperty("ingredients", out var ingredients);
+            structuredJson.RootElement.TryGetProperty("steps", out var steps);
+            structuredJson.RootElement.TryGetProperty("bake_time", out var bakeTime);
+            #endregion
+
+            Assert.AreEqual(ingredients.ValueKind, JsonValueKind.Array);
+            Assert.AreEqual(steps.ValueKind, JsonValueKind.Array);
+            foreach (JsonElement stepElement in steps.EnumerateArray())
+            {
+                stepElement.TryGetProperty("ingredients", out var stepIngredients);
+                stepElement.TryGetProperty("directions", out var stepDirections);
+                Assert.AreEqual(stepIngredients.ValueKind, JsonValueKind.Array);
+                Assert.AreEqual(stepDirections.ValueKind, JsonValueKind.String);
+            }
+            Assert.AreEqual(bakeTime.ValueKind, JsonValueKind.String);
+
+            #region Snippet:Azure_AI_Inference_SampleStructuredOutputPrintOutput
+            var options = new JsonSerializerOptions
+            {
+                WriteIndented = true
+            };
+            Console.WriteLine($"Ingredients: {System.Text.Json.JsonSerializer.Serialize(ingredients, options)}");
+            Console.WriteLine($"Steps: {System.Text.Json.JsonSerializer.Serialize(steps, options)}");
+            Console.WriteLine($"Bake time: {System.Text.Json.JsonSerializer.Serialize(bakeTime, options)}");
+            #endregion
+        }
+
+        private class OverrideApiVersionPolicy : HttpPipelinePolicy
```

Comment:
Is this used anywhere? Can it be removed?

Reply:
Yes, it's used for the endpoint I'm hitting, line 34